moveit_planners/pilz_industrial_motion_planner/src/trajectory_functions.cpp

```diff
@@ -53,11 +53,19 @@ bool pilz_industrial_motion_planner::computePoseIK(const planning_scene::Plannin
     ROS_ERROR_STREAM("Robot model has no planning group named as " << group_name);
     return false;
   }
-
-  if (!robot_model->getJointModelGroup(group_name)->canSetStateFromIK(link_name))
+  const moveit::core::JointModelGroup* jmg = robot_model->getJointModelGroup(group_name);
+  std::string ik_link_name = link_name;
+  Eigen::Isometry3d ik_link_pose = pose;
+  if (jmg->canSetStateFromIK(link_name))
   {
-    ROS_ERROR_STREAM("No valid IK solver exists for " << link_name << " in planning group " << group_name);
-    return false;
+    /// Transforms the goal into a link goal if link_name is a subframe
+    if (scene->knowsFrameTransform(link_name)) {
+      ik_link_name = jmg->getSolverInstance()->getTipFrame();
+      ik_link_pose = pose * scene->getFrameTransform(link_name).inverse() * scene->getFrameTransform(ik_link_name);
```

Comment:
This should check that the transform between `link_name` and `ik_link_name` is _fixed_.

Reply:
Are there methods available in MoveIt I can use to check that?

Reply:
There is `RobotState::getRigidlyConnectedParentLinkModel` which also directly provides the relative transform:
https://github.com/ros-planning/moveit/blob/984a0ea2a00b4190cad9e08ea0d764c19dbf9975/moveit_core/robot_state/include/moveit/robot_state/robot_state.h#L1388-L1390

Reply:
I just noticed that the logic on [this](https://github.com/ros-planning/moveit/blob/c7c1b5e83661327a165196d31bf86b66d6f650c9/moveit_planners/pilz_industrial_motion_planner/src/trajectory_functions.cpp#L59) line should be negated for catching subframes, meaning the fixes to the link name validation logic and `computePoseIK()` seem to be sufficient for planning with subframes: `setFromIK()` already handles the transformation under the hood.

Reply:
Done :+1: